paasta_tools/mesos/master.py

```diff
@@ -69,15 +69,21 @@ def host(self):
         return "{}://{}".format(self.config["scheme"], self.resolve(self.config["master"]))
 
     @log.duration
-    def fetch(self, url, **kwargs):
+    def request(self, url, method=requests.get, **kwargs):
         try:
-            return requests.get(
+            return method(
                 urljoin(self.host, url),
                 timeout=self.config["response_timeout"],
                 **kwargs)
         except requests.exceptions.ConnectionError:
             raise exceptions.MasterNotAvailableException(MISSING_MASTER.format(self.host))
 
+    def fetch(self, url, **kwargs):
```

Comment:
if you're doing this for api compatibility, then remove the default option from ``self.request`` and just set it here instead.